src/shell/llvm_executor.cpp

```diff
@@ -0,0 +1,285 @@
+    TheModule->setDataLayout(J->getDataLayout());
+
+    // Create new pass and analysis managers.
+    auto TheFPM = std::make_unique<FunctionPassManager>();
+    auto TheLAM = std::make_unique<LoopAnalysisManager>();
+    auto TheFAM = std::make_unique<FunctionAnalysisManager>();
+    auto TheCGAM = std::make_unique<CGSCCAnalysisManager>();
+    auto TheMAM = std::make_unique<ModuleAnalysisManager>();
+    auto ThePIC = std::make_unique<PassInstrumentationCallbacks>();
+    auto TheSI = std::make_unique<StandardInstrumentations>(*Context, /*DebugLogging*/ true);
+    TheSI->registerCallbacks(*ThePIC, TheMAM.get());
+
+    // Add transform passes.
+    // Do simple "peephole" optimizations and bit-twiddling optzns.
+    TheFPM->addPass(InstCombinePass());
+    // Reassociate expressions.
+    TheFPM->addPass(ReassociatePass());
+    // Eliminate Common SubExpressions.
+    TheFPM->addPass(GVNPass());
+    // Simplify the control flow graph (deleting unreachable blocks, etc).
+    TheFPM->addPass(SimplifyCFGPass());
+
+    // Register analysis passes used in these transform passes.
+    PassBuilder PB;
+    PB.registerModuleAnalyses(*TheMAM);
+    PB.registerFunctionAnalyses(*TheFAM);
+    PB.crossRegisterProxies(*TheLAM, *TheFAM, *TheCGAM, *TheMAM);
+
+    auto byteptr = builder->getPtrTy();
+        //llvm::Type::getInt8Ty(*Context)->getPo;
+    auto FunctionType = llvm::FunctionType::get(llvm::Type::getVoidTy(*Context), {byteptr, byteptr}, false);
+
+    auto fun = TheModule->getOrInsertFunction("execute", FunctionType);
+
+    Function* F = Function::Create(FunctionType, Function::ExternalLinkage, "__anon_expr", TheModule.get());
+
+    // Add a basic block to the function. As before, it automatically inserts
+    // because of the last argument.
+    BasicBlock* BB = BasicBlock::Create(*Context, "EntryBlock", F);
+
+    builder->SetInsertPoint(BB);
+
+    // Get pointers to the constant `1'.
+
+    const auto [prog, args] = SeparateProg(input);
+
+    auto CalRes = builder->CreateCall(fun,{builder->CreateGlobalString(prog),builder->CreateGlobalString(args)});
+
+    // Value *One = builder.getInt32(1);
+    // Value *Add = builder.CreateAdd(One, One);
+
+    builder->CreateRet(builder->getInt32(1));
+
+    auto RT = J->getMainJITDylib().createResourceTracker();
+
+    auto M = ThreadSafeModule(std::move(TheModule), std::move(Context));
+
+    M.getModuleUnlocked()->print(llvm::outs(), nullptr);
+
+    ExitOnErr(J->addModule(std::move(M), RT));
+
+    // Look up the JIT'd function, cast it to a function pointer, then call it.
+    auto ExprSymbol = ExitOnErr(J->lookup("__anon_expr"));
+
+    void (*FP)() = ExprSymbol.getAddress().toPtr<void (*)()>();
+    FP();
+
+    return EXIT_SUCCESS;
+}
+
+export namespace llvm_backend
+{
+
+int executeLLVM(std::string const& lineBuffer)
+{
+    try
+    {
+        return jitPart(lineBuffer);
+    }
+    catch (std::exception const& e)
+    {
+        std::cout << "Exception caught: " << e.what();
```

Comment:
later on, we should bubble back errors in a way that does not require `cout` to work (imagine unit tests) :)